pkg/plugins/plugins.go

```diff
@@ -166,10 +164,18 @@ func ReadPluginJSON(reader io.Reader) (JSONData, error) {
 	switch plugin.ID {
 	case "grafana-piechart-panel":
 		plugin.Name = "Pie Chart (old)"
-	case "grafana-pyroscope-datasource": // rebranding
-		plugin.Alias = "phlare"
-	case "debug": // panel plugin used for testing
-		plugin.Alias = "debugX"
+	case "grafana-pyroscope-datasource":
+		fallthrough
+	case "annolist":
+		fallthrough
+	case "debug":
+		if plugin.Alias == "" {
+			return plugin, fmt.Errorf("expected alias to be set")
+		}
+	default:
+		if plugin.Alias != "" {
+			return plugin, ErrUnsupportedAlias
```

Comment:
@wbrowne 🤔 actually this is the only condition that needs to stay until we have something upstream in our publish pipeline to block it.  Are you comfortable moving forward with this, then we open a new PR to remove this error after we are confident we won't accidentally publish plugins with alias?